From 90e285139ff4092e2e8d1808f024d0698f12b23a Mon Sep 17 00:00:00 2001
From: Johnny Robeson <johnny@localmomentum.net>
Date: Sat, 20 Dec 2014 21:20:53 -0500
Subject: [PATCH] use RepoQuery in Repo  get(Text|int)Pragma

It seems that SQLite doesn't like reading the value of journal_mode
in RepoTxnQuery, immediately after setting it.

The documentation for journal_mode only says:

"Note also that the journal_mode cannot be changed while a transaction is active."
http://www.sqlite.org/pragma.html#pragma_journal_mode
---
 hphp/runtime/vm/repo.cpp | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/hphp/runtime/vm/repo.cpp b/hphp/runtime/vm/repo.cpp
index b8cc0ec..8590c17 100644
--- a/hphp/runtime/vm/repo.cpp
+++ b/hphp/runtime/vm/repo.cpp
@@ -717,10 +717,9 @@ void Repo::getIntPragma(int repoId, const char* name, int& val) {
   ssPragma << "PRAGMA " << dbName(repoId) << "." << name << ";";
   RepoStmt stmt(*this);
   stmt.prepare(ssPragma.str());
-  RepoTxnQuery query(txn, stmt);
+  RepoQuery query(stmt);
   query.step();
   query.getInt(0, val);
-  txn.commit();
 }
 
 void Repo::setIntPragma(int repoId, const char* name, int val) {
@@ -746,12 +745,11 @@ void Repo::getTextPragma(int repoId, const char* name, std::string& val) {
   ssPragma << "PRAGMA " << dbName(repoId) << "." << name << ";";
   RepoStmt stmt(*this);
   stmt.prepare(ssPragma.str());
-  RepoTxnQuery query(txn, stmt);
+  RepoQuery query(stmt);
   const char* s;
   query.step();
   query.getText(0, s);
   val = s;
-  txn.commit();
 }
 
 void Repo::setTextPragma(int repoId, const char* name, const char* val) {
